target/linux/bcm63xx/base-files/lib/upgrade/platform.sh

```diff
@@ -13,4 +23,50 @@ platform_check_image() {
 	esac
 }
 
-# use default for platform_do_upgrade()
+cfe_jffs2_upgrade_tar() {
+	local tar_file="$1"
+	local kernel_mtd="$(find_mtd_index $CI_KERNPART)"
+
+	if [ ! "$kernel_mtd" ]; then
+		echo "$CI_KERNPART partition not found"
+		return 1
+	fi
+
+	local board_dir=$(tar tf $tar_file | grep -m 1 '^sysupgrade-.*/$')
+	board_dir=${board_dir%/}
+
+	local kernel_length=`(tar xf $tar_file ${board_dir}/kernel -O | wc -c) 2> /dev/null`
+	local rootfs_length=`(tar xf $tar_file ${board_dir}/root -O | wc -c) 2> /dev/null`
```

Comment:
Consider replacing the backticks by $() syntax:

```
	local kernel_length=$((tar xf $tar_file ${board_dir}/kernel -O | wc -c) 2> /dev/null)
	local rootfs_length=$((tar xf $tar_file ${board_dir}/root -O | wc -c) 2> /dev/null)
```

We've done that for other files maybe half a year ago IIRC.

Reply:
Ok, I change it.
FYI, this was copied from: https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/nand.sh#L252

Reply:
Thanks, will send a patch as well. I think back then the changes were limited to target directory.

Reply:
After testing your suggestion it doesn't work, so I'm reverting this change.

Reply:
Sorry, it's kind of a deja vu:
https://github.com/openwrt/openwrt/commit/291d79935e9c4a974b037b6bed733ce92d4242e3#diff-a6a780f4082e88756e922a1f3ec5e2ab

"$((" should be replaced by "$( (", however, in the references PR it was decided to drop the subshell entirely. Sorry for the wrong hint, now I will go and fix my patches just sent :-(